src/test/scala/uk.gov.nationalarchives.fileformat/FFIDExtractorTest.scala

```diff
@@ -0,0 +1,52 @@
+package uk.gov.nationalarchives.fileformat
+
+import java.util.UUID
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.scalatest.matchers.should.Matchers._
+import uk.gov.nationalarchives.aws.utils.SQSUtils
+import uk.gov.nationalarchives.fileformat.FFIDExtractor.FFIDFile
+
+import scala.jdk.CollectionConverters._
+
+
+class FFIDExtractorTest extends FileSpec {
+
+  "The ffid method" should "return the correct droid and signature version" in {
+    val result = FFIDExtractor(sqsUtils, config("result_some_parent_ids")).ffidFile(ffidFile)
+    result.map(r => {
```

Comment:
What happens if `ffidFile` returns an error? Will the test still pass because we're using `map`?

If so, does ScalaTest have a good way to inspect the properties of the `Right` value? Even calling `result.right.get` and letting the test throw an exception would let the test fail, though the error message wouldn't be very informative.

Reply:
Yeah, it will pass if there's an error. I've put a line in before `map` to check `result.isRight should be(true)` which should fail the test if there's an error and should be obvious what it is.

Reply:
I should have said - the same goes for the other tests which use `map`.

If you're checking `isRight`, it might actually be better to use `result.right.get` rather than `map` so that it's clear that the test is checking that specific result.

Reply:
Ah I see. I've changed them over to result.right.get